### DISCUSSION TITLE
Customized format unmarshaller does not work for number type

### DISCUSSION THREAD

**Discussion by YitingTinaRen** [p1c2u/openapi-core#827] [Q&A]
Hello!
I am managing to use this tool to replace the serializer function in the django.
I have the following spec:
```yaml
cost:
    type: number
    format: decimal
```

And the customized format unmarshaller is defined as follows:
```python
from decimal import Decimal

def float_to_decimal(value):
    """
    turn float format into Decimal object
    """
    return Decimal(str(value))


customized_extra_format_unmarshallers = {
    'decimal': float_to_decimal,
}

config = Config(extra_format_unmarshallers=customized_extra_format_unmarshallers)
OpenAPI.from_file_path('index.yaml', config=config)
```

But openapi-core ignores my customized_extra_format_unmarshallers, I would like the cost be a decimal object but it still return float data, is this a bug or is there any error in my setup? It would be helpful if there can be a workaround. 

Thank you!

**Comment by p1c2u** [comment]
Please check https://github.com/python-openapi/openapi-core/issues/821

**Reply by YitingTinaRen** [reply]
Hello @p1c2u,

Thanks a lot for the response!
I checked the 0.19.1 release, and the issue in #821 was indeed solved.
Unfortunately the question in this discussion still exists.
I checked the source code, under openapi_core.unmarshalling.schemas.unmarshallers.SchemaUnmarshaller.find_format
```python
for schema in self.schema_validator.iter_valid_schemas(value):
    schema_validator = self.schema_validator.evolve(schema)
    primitive_type = schema_validator.get_primitive_type(value)

    if primitive_type != "string":
        continue

    if "format" in schema:
        return str(schema.getkey("format"))

return None
```
I believe it is the `if primitive_type != "string":` condition makes the **number** type data can not be processed by the extra_format_unmarshaller defined by users.

I am wondering will the function be open in the future?

BTW many thanks for releasing the solution for #821 in short period!

**Reply by p1c2u** [reply]
The condition was set to simplify the process otherwise we would have to pass type for each format unmarshaller. I hope in future we can support all types.
